src/hooks/interfaces.ts

```diff
@@ -0,0 +1,31 @@
+import { IFormatOptionQuery } from '@/utils/getPagiantion';
+import { UseQueryOptions } from 'react-query';
+
+export interface PaginationServicer<T> {
+  getPagination: (query: IFormatOptionQuery) => Promise<T>;
+}
+
+export interface Pagination<T> {
```

Comment:
It would be clearer if we name `IAPIResponse` instead of `Pagination`.

Reply:
API respone many interface not only  Pagination , so if we change Pagination -> IAPIResponse make confuse for who use it before